### PR TITLE
devcontainer bug fix

### PR DESCRIPTION
**Pull request by CallMeGreg**
This pull request includes changes to the development environment setup and dependency updates.

The most important changes are the removal of manual setup for dotnet and Node.js in the Dockerfile, addition of features in `devcontainer.json`, introduction of Dependabot configuration, and updates to the versions of `@sveltejs/adapter-auto`, `@sveltejs/kit`, and `vite` in `package.json`.

Edit: includes super-linter error fixes as well

### CODE REVIEW

File: admin/README.md [hunk 1]
```diff
@@ -17,7 +17,9 @@ Because GitHub Copilot is a coding tool, we wanted to provide a scenario which r
 - A challenge flexible enough to allow for multiple languages to be available for learners to use.
 - Something which has enough substance for learners to build, explore, and use real-world techniques and strategies.
 
-To support this style, we've created a workshop in the style of a hackathon. Learners are given a set of [challenges](./content/) based around a [dataset with US flight information](../data/). The "green path" involves learners creating a model to predict the chance a flight will be delayed more than 15 minutes based on the day of the week and the destination, export the model and a list of airports to expose as an API, and a frontend for users to access the mode.
+To support this style, we've created a workshop in the style of a hackathon. Learners are given a set of [challenges](./content/) based around
+a [dataset with US flight information](../data/). The "green path" involves learners creating a model to predict the chance a flight will be delayed
+more than 15 minutes based on the day of the week and the destination, export the model and a list of airports to expose as an API, and a frontend for users to access the mode.
```
Comment (gclhub) [inline admin/README.md]:
```suggestion
more than 15 minutes based on the day of the week and the destination, export the model and a list of airports to expose as an API, and a frontend for users to access the model.
```

File: content/1-create-model-data.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Working with a dataset and machine learning
 
-The scenario for this workshop revolves around a [dataset from the FAA](../data/flights.csv) which contains core information about flights in the US in 2013. It contains the carrier, the date and time the flights took off and arrived (and their origin and destination), and information about delays including both the time and a flag if it was more than 15 minutes. For this scenario you will explore and cleanse the data, build an export a model trained to show the percent chance a flight will be delayed for a day of the week and an airport, and create a new CSV file containing the list of all airports and their associated id in the dataset.
+The scenario for this workshop revolves around a [dataset from the FAA](../data/flights.csv) which contains core information about flights in the US in 2013.
+It contains the carrier, the date and time the flights took off and arrived (and their origin and destination), and information about delays including both
+the time and a flag if it was more than 15 minutes. For this scenario you will explore and cleanse the data, build an export a model trained to show the percent chance
```
Comment (gclhub) [inline content/1-create-model-data.md]:
```suggestion
the time and a flag if it was more than 15 minutes. For this scenario you will explore and cleanse the data, build and export a model trained to show the percent chance
```

File: content/1-create-model-data.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
+a flight will be delayed for a day of the week and an airport, and create a new CSV file containing the list of all airports and their associated id in the dataset.
```
Comment (gclhub) [inline content/1-create-model-data.md]:
```suggestion
a flight will be delayed for a day of the week and an airport, and create a new CSV file containing the list of all airports and their associated IDs in the dataset.
```

File: content/1-create-model-data.md [hunk 4]
```diff
@@ -15,9 +18,13 @@ You will have successfully completed this challenge after:
 
 ## Tips
 
-If you're new to data science and machine learning, the first step when working with a new dataset is to explore the data. You'll typically start by loading the data into memory, displaying a portion to gain an understanding about how the data is structured, and looking for any issues which need to be cleaned up. (There are always issues which need to be cleaned up.) As these are common tasks, you'll find GitHub Copilot will be able to provide a fair amount of support.
+If you're new to data science and machine learning, the first step when working with a new dataset is to explore the data.
+You'll typically start by loading the data into memory, displaying a portion to gain an understanding about how the data is structured,
```
Comment (gclhub) [inline content/1-create-model-data.md]:
```suggestion
You'll typically start by loading the data into memory, displaying a portion of it to gain an understanding about how the data is structured,
```

File: content/3-create-frontend.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # Create a user experience
 
-With the API created it's time to close everything out by creating a frontend application. While the first two exercises have been relatively prescriptive in the frameworks and style, this exercise offers you the ability to choose the direction you'd like to go for creating a frontend. You can create a website using React or Svelte, a mobile app, a Windows app... Whatever you feel most comfortable with! You can also use this as an opportunity to explore a new framework.
+With the API created it's time to close everything out by creating a frontend application. While the first two exercises have been relatively prescriptive in the frameworks and style,
+this exercise offers you the ability to choose the direction you'd like to go for creating a frontend. You can create a website using React or Svelte, a mobile app, a Windows app...
```
Comment (gclhub) [inline content/3-create-frontend.md]:
```suggestion
this exercise offers you the ability to choose the direction you'd like to go for creating a frontend. You can create a website using React or Svelte, a mobile app, a Windows app, or
```

File: content/3-create-frontend.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
...
+Whatever you feel most comfortable with! You can also use this as an opportunity to explore a new framework.
```
Comment (gclhub) [inline content/3-create-frontend.md]:
```suggestion
whatever you feel most comfortable with! You can also use this as an opportunity to explore a new framework.
```

File: hackathon.md [hunk 7]
```diff
@@ -1,20 +1,26 @@
...
 
 ## The scenario
 
-You have been provided a [dataset with flight information from the FAA](./data/flights.csv). The dataset contains dates, times and carriers for flights in the US which took place in 2013. You want to build an application which will allow someone to select the day of the week and arrival airport to see the chance their flight will be delayed by more than 15 minutes. You'll do so by walking through the following challenges:
+You have been provided a [dataset with flight information from the FAA](./data/flights.csv). The dataset contains dates,
+times and carriers for flights in the US which took place in 2013. You want to build an application which will allow someone to select the day of the week and
+arrival airport to see the chance their flight will be delayed by more than 15 minutes. You'll do so by walking through the following challenges:
 
 1. Create and export the data and model to support the application
 2. Create an API to provide a list of airports and their associated ID in the data, and the model
```
Comment (gclhub) [inline hackathon.md]:
```suggestion
2. Create an API to provide a list of airports and their associated IDs in the data, and the model
```

File: microsoft.asc
Comment (gclhub) [inline microsoft.asc:1]: Is this file and the .gpg encrypted one supposed to be part of the commit?

**Review by gclhub — APPROVED** [review]
Some miscellaneous verbiage suggestions but LGTM otherwise 🚀

File: microsoft.asc
Comment (gclhub) [reply microsoft.asc:1]: Never mind, it looks like it's needed to access the devcontainers features?

Comment (GeekTrainer) [reply microsoft.asc:1]: Yeah, I'm not sure why they wound up being generated either, but they make things work so I'm not going to touch it 😆

**Review by ppremk — APPROVED** [review]

**Pull request merged**